calico-enterprise/operations/monitor/metrics/recommended-metrics.mdx

```diff
@@ -0,0 +1,514 @@
+- [Felix error metrics](#felix-error-metrics)
+- [Felix time-based metrics](#felix-time-based-metrics)
+
+## Policy metrics
+
+### Denied traffic
+
+| Denied traffic                   |                                                              |
+| -------------------------------- | ------------------------------------------------------------ |
+| Metric                           | *calico_denied_packets / calico_denied_bytes*                |
+| Explanation                      | Number of packets or bytes that have been dropped by explicit or implicit deny rules. |
+| Threshold value recommendation   | The general rule of thumb is this metric should report zero at a stable state. Any deviation means that policy and traffic have diverged. Achieving a zero state depends on the stability and maturity of your cluster and policy. |
+| Threshold breach symptoms        | Either unexpected traffic is being denied because of an attack (one example), or expected traffic is being denied because of a misconfiguration in a policy. |
+| Threshold breach recommendations | If this metric indicates that policy and traffic have diverged, the recommended steps are: Determine if an attack is causing the metric to spike, or if these flows should be allowed. If the flow should indeed be allowed, update the policy or a preceding policy to allow this traffic. |
+| Priority level                   | Recommended.                                                 |
+
+### Traffic per rule
+
+| Traffic per rule                 |                                                              |
+| -------------------------------- | ------------------------------------------------------------ |
+| Metric                           | *cnx_policy_rule_bytes / cnx_policy_rule_packets*            |
+| Explanation                      | Number of bytes or packets handled by {{prodname}} network policy rules. |
+| Threshold value recommendation   | This metric should usually be non-zero (unless expected).  A zero value indicates the rule is not matching any packets, and could be surplus to requirements. |
+| Threshold breach symptoms        | N/A                                                          |
+| Threshold breach recommendations | If this metric consistently reports a zero value over an acceptable time, you can consider removing the policy rule. |
+| Priority Level                   | Optional.                                                    |
+
+
+## Typha general metrics
+
+### Data store cache size
+
+| Data store cache size             |                                                              |
+| -------------------------------- | ------------------------------------------------------------ |
+| Metric                           | *max by (instance) <(typha_cache_size>*<br />typha_cache_size[syncer="felix"]<br/>typha_cache_size[syncer="bgp"] <br/>typha_cache_size[syncer="dpi"]<br />typha_cache_size[syncer="tbd"] |
```

Comment:
A lot of the metrics are getting garbled when converted to HTML, I think they all need to be in code blocks.  Anything with a `<...>` in it gets broken by markdown because it looks like an inline HTML tag.

I feel like we're going to confuse people by presenting `max by (instance) <(typha_cache_size>` as a "metric".  It's actually a Prometheus query over the metrics.

In this case, Typha runs multiple copies of the cache, one for each client type.  So it emits multiple cach-size metrics, one for each copy.  I feel like what we should do is: explain the raw metrics and then suggest a useful query to turn the raw metrics into actionable insights.  If we just present the queries then I think users will feel like it's a non-sequitur.

Reply:
Changed all metrics to code block. Copied updates from gdoc for datastore cache size